### PR TITLE
ci: separarte pull-containerd-node-e2e for 1.5 branch

### PR DESCRIPTION
**Pull request by akhilerm**
containerd v1.5.x supports CRI v1alpha2, the API that was available at the time of release for containerd v1.5.
containerd v1.6.x has support for both CRI v1alpha2 and v1; and is being designated a long term support release.

kubelet master is removing support for CRI v1alpha2, this action has the effect of forcing kubernetes master(and kubernetes r.next+) users to move up to containerd v1.6.x where both CRI v1 and v1alpha2 is supported.

Therefore we need to separate out the `pull-containerd-node-e2e` job for containerd 1.5 branch, so that patches can still be made to 1.5 branch till its EOL. Instead of running against kubernetes master, it will run against k8s release-1.25 branch (the last release which supports CRI v1alpha2)

Ref: https://github.com/kubernetes/kubernetes/pull/110618

Signed-off-by: Akhil Mohan <makhil@vmware.com>

### CODE REVIEW

**Comment by akhilerm** [comment]
/cc @dims @adisky @mikebrow 

We need this job to run till containerd 1.5 EOL, so that the 1.5 branch can still accept patches

**Comment by mikebrow** [comment]
nit on the explanation text in the PR...
containerd v1.5.x supports CRI v1alpha2, the api that was available at the time of release for containerd v.1.5
containerd v1.6.x has support for both CRI v1alpha2 and v1; and is being designated a long term support release

kubelet master is removing support for CRI v1alpha2, this action has the effect of forcing kubernetes master(and kubernetes r.next+) users to move up to containerd v1.6.x where both CRI v1 and v1alpha2 is supported...

**Review by samuelkarp — APPROVED** [review]
I'm not super familiar with the configuration here, but this LGTM.

**Review by qiutongs — APPROVED** [review]
Thanks for the fix.

**Comment by akhilerm** [comment]
@mikebrow Updated the PR description

**Review by adisky — COMMENTED** [review]
Thanks @akhilerm 
/lgtm

**Review by mikebrow — APPROVED** [review]
LGTM

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml
Comment (bobbypage) [inline config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:129]: This image config is pointing to to `cos-stable` (https://github.com/kubernetes/test-infra/blob/master/jobs/e2e_node/containerd/containerd-main-presubmit/image-config-presubmit.yaml#L3-L5)

cos-stable is using COS 101 which has cgroupv2 enabled. cgroupv2 should enable the systemd cgroup driver. 

I think we should also add https://github.com/kubernetes/test-infra/blob/master/jobs/e2e_node/containerd/containerd-main/cgroupv2/env-cgroupv2#L10 in `CONTAINERD_SYSTEMD_CGROUP: 'true'`

and add `--cgroup-driver=systemd` to kubelet flags like https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/containerd.yaml#L238

Comment (akhilerm) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:129]: Since all the images for containerd are updated to `cos-stable`, should we add the cgroup_env to all the tests except `containerd-main/cgroupv1`?

Comment (bobbypage) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:129]: Yup, that makes sense.

Comment (akhilerm) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:129]: I have one more doubt related to the cgroupv2, these env is currently not set on the image files for other tests, but its still passing. So isnt it a mandatory requirement?

Comment (bobbypage) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:129]: Not mandatory (if it is not specified it will use the cgroupfs driver). But the systemd driver is what is recommended to be used with cgroupv2 (https://kubernetes.io/docs/concepts/architecture/cgroups/#requirements).

We want to make it default in future version of containerd - https://github.com/containerd/containerd/issues/7319

Comment (akhilerm) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:129]: Shouldnt we also add the `CONTAINERD_COS_CGROUP_MODE: 'v2'` to the env? Or is it set by default in cos-101?

File: jobs/e2e_node/containerd/containerd-main-presubmit/env
Comment (akhilerm) [inline jobs/e2e_node/containerd/containerd-main-presubmit/env:9]: @bobbypage Should we add the `CONTAINERD_COS_CGROUP_MODE: 'v2'` also here?

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml
Comment (adisky) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml]: @akhilerm I think its fine not to set `CONTAINERD_COS_CGROUP_MODE: 'v2'` as cgroups v2 is enabled by default in cos 101

@bobbypage @mikebrow so this pull job will cover the cgroupsv2 only, do we need to add an additional pull job for cgroupsv1?

**Review by endocrimes — APPROVED** [review]
/approve
^
node e2e change

I'll defer to @bobbypage  on the COS_CGROUP_MODE thing.

/hold
^
unhold when ready

**Review by mikebrow — APPROVED** [review]
LGTM

**Comment by SergeyKanzhelev** [comment]
/unhold

lgtm

@bobbypage lmk if something needs to be changed

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml
Comment (bobbypage) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml]: Yes, it's fine to not set `CONTAINERD_COS_CGROUP_MODE`, this is only needed to force COS to enable cgroupv2 but since it's already enabled started from COS97 this is unnecessary

**Comment by bobbypage** [comment]
> /unhold
> 
> lgtm
> 
> @bobbypage lmk if something needs to be changed

Yes, we need to ensure that the systemd cgroup driver is configured for all the containerd jobs using COS M97+. Looks like this will be taken care of in https://github.com/kubernetes/test-infra/pull/27943, thanks @akhilerm for followup PR.

**Pull request merged**
